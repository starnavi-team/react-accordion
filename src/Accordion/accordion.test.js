import React from 'react';
import { shallow } from 'enzyme';
import {
  Accordion,
  AccordionGroup,
  AccordionGroupTitle,
  AccordionGroupBody,
} from '../index';
import { StyledAccordion } from './index';

describe('Accordion', () => {
  it('Makes possible to set custom className', () => {
    const testAccordion = shallow(
      <Accordion className="testing">
        <AccordionGroup>
          <AccordionGroupTitle />
          <AccordionGroupBody />
        </AccordionGroup>
      </Accordion>,
    );
    expect(testAccordion.hasClass('testing')).toBe(true);
  });

  it('Makes possible to set custom styles', () => {
    const testAccordion = shallow(
      <Accordion className="testing" style={{ color: 'black' }}>
        <AccordionGroup>
          <AccordionGroupTitle />
          <AccordionGroupBody />
        </AccordionGroup>
      </Accordion>,
    );
    expect(testAccordion.find('.testing')).toHaveStyle('color', 'black');
  });

  it('styledAccordion receives custom className', () => {
    const styled = shallow(<StyledAccordion className="tested" />);
    expect(styled.hasClass('tested')).toBe(true);
  });

  it('styledAccordion receives custom styles', () => {
    const styled = shallow(<StyledAccordion className="tested" style={{ color: 'yellow' }} />);
    expect(styled.find('.tested')).toHaveStyle('color', 'yellow');
  });

  it('should render correctly', () => {
    const component = shallow(<Accordion />);
    expect(component).toMatchSnapshot();
  });
});
