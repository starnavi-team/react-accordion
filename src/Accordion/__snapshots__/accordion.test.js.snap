// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion should render correctly 1`] = `<styled.div />`;
